### PR TITLE
[BE] Migrate codebook unit test to pytest

### PR DESCRIPTION
**Pull request by langong347**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #264
* #257
* #263
* __->__ #259
* #262
* #258
* #256
* #255
* #254

Summary:
- `unittest` to `pytest`
- Use `assert_expected_wrapper` for `CodebookOutput` value assertion

Test Plan:
```
$ python -m pytest --cov=torchmultimodal/modules/layers/ test/modules/layers/test_codebook.py -vv
================================================= test session starts ==================================================
platform darwin -- Python 3.8.13, pytest-7.1.2, pluggy-1.0.0 -- /Users/langong/local/miniconda3/envs/t2v/bin/python
cachedir: .pytest_cache
rootdir: /Users/langong/gpt_attention, configfile: pyproject.toml
plugins: mock-3.8.2, cov-3.0.0
collected 11 items

test/modules/layers/test_codebook.py::TestCodebook::test_quantized_output PASSED                                 [  9%]
test/modules/layers/test_codebook.py::TestCodebook::test_preprocess PASSED                                       [ 18%]
test/modules/layers/test_codebook.py::TestCodebook::test_preprocess_channel_dim_assertion PASSED                 [ 27%]
test/modules/layers/test_codebook.py::TestCodebook::test_postprocess PASSED                                      [ 36%]
test/modules/layers/test_codebook.py::TestCodebook::test_init_embedding PASSED                                   [ 45%]
test/modules/layers/test_codebook.py::TestCodebook::test_ema_update_embedding PASSED                             [ 54%]
test/modules/layers/test_codebook.py::TestCodebook::test_register_buffer_tensors PASSED                          [ 63%]
test/modules/layers/test_codebook.py::TestCodebook::test_init_embedding_smaller_encoded PASSED                   [ 72%]
test/modules/layers/test_codebook.py::TestCodebook::test_codebook_restart PASSED                                 [ 81%]
test/modules/layers/test_codebook.py::TestCodebook::test_load_state_dict PASSED                                  [ 90%]
test/modules/layers/test_codebook.py::TestCodebook::test_lookup PASSED                                           [100%]

---------- coverage: platform darwin, python 3.8.13-final-0 ----------
Name                                                   Stmts   Miss  Cover
--------------------------------------------------------------------------
torchmultimodal/modules/layers/codebook.py                87      1    99%
--------------------------------------------------------------------------
TOTAL                                                    489    403    18%


================================================== 11 passed in 1.56s ============
```

Differential Revision: [D38642049](https://our.internmc.facebook.com/intern/diff/D38642049)

### CODE REVIEW

File: test/modules/layers/test_codebook.py [hunk 1]
```diff
@@ -4,250 +4,306 @@
...
+    actual = model.codebook.code_usage
+    expected = state_dict["codebook.code_usage"]
+    assert_expected(actual, expected)
+
+    actual = model.codebook.code_avg
+    expected = state_dict["codebook.code_avg"]
+    assert_expected(actual, expected)
+
+
+def test_lookup(codebook, embedding_weights):
```
Comment (RdoubleA) [inline test/modules/layers/test_codebook.py:306]: are theses remnants of the next PR?

File: test/modules/layers/test_codebook.py [hunk 2]
```diff
@@ -4,250 +4,306 @@
...
 
+import pytest
+
 import torch
 from test.test_utils import assert_expected, set_rng_seed
 from torch import nn, tensor
 from torchmultimodal.modules.layers.codebook import Codebook
 
 
-class TestCodebook(unittest.TestCase):
```
Comment (RdoubleA) [inline test/modules/layers/test_codebook.py:17]: can we still contain all the tests in the `TestCodebook` class?

File: test/modules/layers/test_codebook.py [hunk 3]
```diff
@@ -4,250 +4,306 @@
...
+    expected = state_dict["codebook.embedding"]
+    assert_expected(actual, expected)
+
+    actual = model.codebook.code_usage
+    expected = state_dict["codebook.code_usage"]
+    assert_expected(actual, expected)
+
+    actual = model.codebook.code_avg
+    expected = state_dict["codebook.code_avg"]
+    assert_expected(actual, expected)
```
Comment (RdoubleA) [inline test/modules/layers/test_codebook.py:303]: maybe you could replace these three assertion with the `assert_expected_wrapper` utility?

File: test/modules/layers/test_codebook.py (hunk 1 above)
Comment (langong347) [reply test/modules/layers/test_codebook.py:306]: Yes. I will remove them.

File: test/modules/layers/test_codebook.py (hunk 2 above)
Comment (langong347) [reply test/modules/layers/test_codebook.py:17]: Grouping tests by class becomes a must when the same test script contains multiple modules to test. When there's only one module, I think either way is fine technically.

File: test/modules/layers/test_codebook.py (hunk 3 above)
Comment (langong347) [reply test/modules/layers/test_codebook.py:303]: Ok. I added value assertion to the wrapper. Will rebase and incorporate. Thanks for the reminder.

Comment (langong347) [reply test/modules/layers/test_codebook.py:303]: Actually, `assert_expected_wrapper` is meant for actual being `NamedTuple` but in this block we are dealing with codebook parameters.

I have changed other places wherever applicable.

**Comment by langong347** [comment]
@langong347 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38642049).

File: test/modules/layers/test_codebook.py (hunk 3 above)
Comment (RdoubleA) [reply test/modules/layers/test_codebook.py]: oh yeah I missed that, makes sense

**Review by RdoubleA — APPROVED** [review]

**Comment by langong347** [comment]
@langong347 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38642049).

**Pull request closed without merging**